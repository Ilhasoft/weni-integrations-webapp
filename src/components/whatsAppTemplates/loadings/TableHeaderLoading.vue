<template>
  <div class="loading">
    <unnnic-skeleton-loading width="40px" height="40px" />
    <unnnic-skeleton-loading width="120px" height="40px" />
  </div>
</template>
<script>
  export default {
    name: 'TableHeaderLoading',
  };
</script>
<style lang="scss" scoped>
  .loading {
    display: flex;
    gap: $unnnic-spacing-inline-md;
  }
</style>
