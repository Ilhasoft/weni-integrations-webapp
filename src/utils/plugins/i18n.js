import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
