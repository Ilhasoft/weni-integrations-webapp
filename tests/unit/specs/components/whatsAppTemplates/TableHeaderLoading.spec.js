import TableHeaderLoading from '@/components/whatsAppTemplates/loadings/TableHeaderLoading.vue';
import { mount, createLocalVue } from '@vue/test-utils';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();

describe('TableHeaderLoading', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(TableHeaderLoading, {
      localVue,
      i18n,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        UnnnicSkeletonLoading: true,
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
