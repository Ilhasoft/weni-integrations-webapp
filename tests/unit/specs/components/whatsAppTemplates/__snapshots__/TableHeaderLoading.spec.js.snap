// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeaderLoading should be rendered properly 1`] = `
<div class="loading">
  <unnnicskeletonloading-stub width="40px" height="40px"></unnnicskeletonloading-stub>
  <unnnicskeletonloading-stub width="120px" height="40px"></unnnicskeletonloading-stub>
</div>
`;
